src/auto/dbus_connection.rs

```diff
@@ -0,0 +1,1264 @@
+        >(
+            connection: *mut gio_sys::GDBusConnection,
+            sender_name: *const libc::c_char,
+            object_path: *const libc::c_char,
+            interface_name: *const libc::c_char,
+            signal_name: *const libc::c_char,
+            parameters: *mut glib_sys::GVariant,
+            user_data: glib_sys::gpointer,
+        ) {
+            let connection = from_glib_borrow(connection);
+            let sender_name: Borrowed<GString> = from_glib_borrow(sender_name);
+            let object_path: Borrowed<GString> = from_glib_borrow(object_path);
+            let interface_name: Borrowed<GString> = from_glib_borrow(interface_name);
+            let signal_name: Borrowed<GString> = from_glib_borrow(signal_name);
+            let parameters = from_glib_borrow(parameters);
+            let callback: &P = &*(user_data as *mut _);
+            (*callback)(
+                &connection,
+                sender_name.as_str(),
+                object_path.as_str(),
+                interface_name.as_str(),
+                signal_name.as_str(),
+                &parameters,
+            );
+        }
+        let callback = Some(callback_func::<P> as _);
+        unsafe extern "C" fn user_data_free_func_func<
+            P: Fn(&DBusConnection, &str, &str, &str, &str, &glib::Variant) + 'static,
+        >(
+            data: glib_sys::gpointer,
+        ) {
+            let _callback: Box_<P> = Box_::from_raw(data as *mut _);
+        }
+        let destroy_call9 = Some(user_data_free_func_func::<P> as _);
+        let super_callback0: Box_<P> = callback_data;
+        unsafe {
+            gio_sys::g_dbus_connection_signal_subscribe(
+                self.to_glib_none().0,
+                sender.to_glib_none().0,
+                interface_name.to_glib_none().0,
+                member.to_glib_none().0,
+                object_path.to_glib_none().0,
+                arg0.to_glib_none().0,
+                flags.to_glib(),
+                callback,
+                Box_::into_raw(super_callback0) as *mut _,
+                destroy_call9,
+            )
+        }
+    }
+
+    pub fn signal_unsubscribe(&self, subscription_id: u32) {
+        unsafe {
+            gio_sys::g_dbus_connection_signal_unsubscribe(self.to_glib_none().0, subscription_id);
+        }
+    }
+
+    pub fn start_message_processing(&self) {
+        unsafe {
+            gio_sys::g_dbus_connection_start_message_processing(self.to_glib_none().0);
+        }
+    }
+
+    pub fn unexport_action_group(&self, export_id: u32) {
+        unsafe {
+            gio_sys::g_dbus_connection_unexport_action_group(self.to_glib_none().0, export_id);
+        }
+    }
+
+    pub fn unexport_menu_model(&self, export_id: u32) {
+        unsafe {
+            gio_sys::g_dbus_connection_unexport_menu_model(self.to_glib_none().0, export_id);
+        }
+    }
+
+    pub fn unregister_object(&self, registration_id: u32) -> bool {
```

Comment:
This should get a newtype wrapper. Similar to `Object::connect()` etc (of course also for the register function the same)